### ISSUE TITLE
When we use multiple scopes with same key, the outermost scope is used

### ISSUE THREAD

**Issue by Andreas-Hjortland** [mattwcole/gelf-extensions-logging#15]
Hi,

I tried to set the key `CorrelationId` using `_logger.BeginScope(("CorrelationId", "my-correlation-id"))`, but it doesn't work because ASP.NET Core sets a correlation id scope on the logger outside of my control. I believe it would be better that if I set a CorrelationId (or any other duplicate keyed scope) the innermost scope should be used instead of the outermost.

I created a failing test which shows what I mean:
```c#
[Fact]
public async Task When_duplicate_scope_keys_inner_scope_should_be_used()
{
    var messageText = _faker.Lorem.Sentence();

    var sut = _loggerFixture.CreateLogger<GelfLoggerTests>();
    using (sut.BeginScope(("foo", "outer")))
    {
        using (sut.BeginScope(("foo", "inner")))
        {
            sut.LogCritical(messageText);
        }
    }

    var message = await _graylogFixture.WaitForMessageAsync();

    Assert.Equal("inner", message.foo); // Fails. message.foo is "outer"
}
```

**Comment by mattwcole** [comment]
I haven't noticed this issue in my ASP.NET Core apps. Perhaps the middleware in which I set the correlation ID is running before the framework sets its value? In that case, allowing the inner scope to override the outer would not be desirable for apps such as mine.

It's also possible that if the framework just sets CorrelationId  to be a GUID I may have not noticed it overriding my own value. I'll take a closer look.

**Comment by mattwcole** [comment]
I wasn't able to find a CorrelationId added by ASP.NET (at least not in the basic sample app in this repo), but I think the change makes sense. I have extended behaviour to allow scope fields to be overridden by structured logs as well.

**Comment by mattwcole** [comment]
Released as [v1.4.0-pre2](https://www.nuget.org/packages/Gelf.Extensions.Logging/1.4.0-pre2).
